plugin/src/main/java/io/fabric8/maven/plugin/SetupMojo.java

```diff
@@ -0,0 +1,268 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.plugin;
+
+import io.fabric8.maven.core.util.MavenUtil;
+import io.fabric8.utils.DomHelper;
+import io.fabric8.utils.Strings;
+import io.fabric8.utils.XmlUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProjectHelper;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Text;
+
+import java.io.File;
+import java.util.Objects;
+
+import static io.fabric8.utils.DomHelper.firstChild;
+import static org.eclipse.jgit.lib.ObjectChecker.parent;
+
+/**
+ * Ensures that the fabric8 maven plugin is defined and setup correctly in the projects pom.xml
+ */
+@Mojo(name = "setup", requiresProject = true)
+public class SetupMojo extends AbstractFabric8Mojo {
+
+    public static final String PLUGIN_GROUPID = "io.fabric8";
+    public static final String PLUGIN_ARTIFACTID = "fabric8-maven-plugin";
+    public static final String PLUGIN_DEFAULT_VERSION = "3.1.18";
+    public static final String FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY = "fabric8.maven.plugin.version";
+
+    /**
+     * Controls whether a backup pom should be created (default is true).
+     */
+    @Parameter(property = "generateBackupPoms")
+    private Boolean generateBackupPoms;
+
+    @Component
+    private MavenProjectHelper projectHelper;
+
+    @Override
+    public void executeInternal() throws MojoExecutionException, MojoFailureException {
+        File pom = project.getFile();
+        boolean updated = false;
+        Document doc;
+        try {
+            doc = XmlUtils.parseDoc(pom);
+        } catch (Exception e) {
+            getLog().error("Failed to parse pom " + pom + ". " + e, e);
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+        if (createOrUpdateFabric8MavenPlugin(doc)) {
+            updated = true;
+        }
+        if (updated) {
+            getLog().info("Updating the pom " + pom);
+            try {
+                DomHelper.save(doc, pom);
+            } catch (Exception e) {
+                getLog().error("Failed to update pom " + pom + ". " + e, e);
+                throw new MojoExecutionException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private boolean createOrUpdateFabric8MavenPlugin(Document doc) {
+        boolean updated = false;
+        String latestVersion = MavenUtil.getVersion(PLUGIN_GROUPID, PLUGIN_ARTIFACTID, PLUGIN_DEFAULT_VERSION);
+
+        // lets check that there's a property defined
+        String currentVersion = project.getProperties().getProperty(FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
```

Comment:
I was hoping we could use the `fabric8:setup` goal to upgrade older versions of the plugin (as updating the plugin version and using the latest executions etc). 

e.g. imagine if a project was using 3.1.1 then we introduced 3.1.29 which now uses the `helm` execution too. If we don't update the version, the project is broken as the new execution isn't available in the old version.

I think I'd rather the `fabric8:setup` add unnecessary versions which folks can remove (due to wishing to define them in a parent pom or via pluginDependency or whatever), than break builds.

As a compromise I've added a flag to disable updating of the plugin version if the plugin is already used. e.g.:
```
mvn io.fabric8:fabric8-maven-plugin:3.1-SNAPSHOT:setup  -DupdateVersion=false
```

will not update the version if its on an older version (or the version is defined somewhere else than the plugin element)